### ISSUE TITLE
Why caffe-style pretrained models need to freeze BN?

### ISSUE THREAD

**Issue by karlind** [open-mmlab/mmdetection#1175]
Why caffe-style pretrained models need to freeze BN but pytorch-style pretrained models don't?

**Comment by hellock** [comment]
`freeze BN` is not precise. You may refer to #346 #437 and our [technical report](https://arxiv.org/abs/1906.07155).

**Comment by karlind** [comment]
I have read related issues and technical report, they explained `eval` and `requires grad` in detail. But they didn't explain why caffe-style pretrained models have different settings in `requires_grad`.
I find all caffe-style backbone set `requires_grad=False`,  while pytorch-style backbone set `requires_grad=True`. I am curious why they have different settings.

**Comment by hellock** [comment]
For caffe-style models, we adopt the pretrained weights converted from detectron, where the BN layers are merged into the preceding conv layer. If you use models pretrained by yourself, you can also set `requires_grad` to True.

**Comment by johnlanbor** [comment]
sorry to reopen this question .

I find detectron pretrained models, bn doesnt have running_mean and running_var parameters. eg https://dl.fbaipublicfiles.com/detectron/ImageNetPretrained/25093814/X-152-32x8d-IN5k.pkl
in mask rcnn benchmark and Detectron2, just set running_mean=0 and running_var=1, If load this pretrained model ,do I need to set this parameters, or can you give a X152 model which have this two parameters?
